### PR TITLE
baseURI

### PR DESCRIPTION
**Pull request by bluelovers**
```ts
import * as cheerio from '..';

let html = `<body><a href="/n2627t/1/"/></body>`;

let $1 = cheerio.load(html);
let $2 = cheerio.load(html, {
	baseURI: 'http://ncode.syosetu.com',
});

console.log($1('a').prop('href')); // => /n2627t/1/
console.log($2('a').prop('href')); // => http://ncode.syosetu.com/n2627t/1/
```

### CODE REVIEW

**Comment by fb55** [comment]
Tests are failing

**Comment by bluelovers** [comment]
hmm i can't do anymore for low version..............
hope someone can make a version for this

**Comment by fb55** [comment]
Would be great if this 1) was optional and 2) had another test case. You probably also want to resolve `src` attributes.

**Comment by bluelovers** [comment]
@fb55 this already is optional if we didn't set baseURI

**Comment by fb55** [comment]
Okay, I see. So whatwg-url depends on node >= 6, while cheerio still supports node@0.12. Can we just use [`url.resolve`](https://nodejs.org/api/url.html#url_url_resolve_from_to) then? Also, would you mind adding a test case for this?

File: lib/api/attributes.js [hunk 1]
```diff
@@ -125,6 +126,13 @@ exports.prop = function (name, value) {
       case 'nodeName':
         property = this[0].name.toUpperCase();
         break;
+      case 'href':
+      case 'src':
+        property = getProp(this[0], name);
```
Comment (fb55) [inline lib/api/attributes.js]: Also, move this into the `if` block?

Comment (bluelovers) [reply lib/api/attributes.js]: what mean for into if?
u mean move this code to case default and make a if block?

but i think is no need

the default is `property = getProp(this[0], name);`

and when we didn't set baseurl
also is  `property = getProp(this[0], name);`

**Comment by bluelovers** [comment]
alright should be ok now

File: lib/api/attributes.js [hunk 2]
```diff
@@ -125,6 +124,19 @@ exports.prop = function (name, value) {
       case 'nodeName':
         property = this[0].name.toUpperCase();
         break;
+      case 'href':
+      case 'src':
+        var allowUrlResolve = {
```
Comment (fb55) [inline lib/api/attributes.js]: Can this be converted to

```js
(name === 'href' && tagName === 'a') ||
(name === 'href' && (tagName === 'img' || tagName === 'iframe'))
```

inside of the `if` clause?

File: lib/api/attributes.js [hunk 3]
```diff
@@ -125,6 +124,19 @@ exports.prop = function (name, value) {
...
+          'href': ['A'],
+          'src': ['IMG', 'IFRAME'],
+        };
+
+        property = getProp(this[0], name);
+
+        if (this.options.baseURI && allowUrlResolve[name] && allowUrlResolve[name].indexOf(this.prop('tagName')) != -1) {
+          property = utils.urlResolve(property, this.options.baseURI);
```
Comment (fb55) [inline lib/api/attributes.js]: Can we use `url.resolve` directly here?

File: lib/api/attributes.js (hunk 2 above)
Comment (fb55) [reply lib/api/attributes.js]: Also, this doesn't cover a bunch of cases, but imho that's okay for now.

**Comment by mmontag** [comment]
This would be nice to have!

File: test/api/url.resolve.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+describe('$(...)', function()
+{
+  var $;
+
+  describe('.prop', function ()
+  {
+    beforeEach(function ()
+    {
+      $ = cheerio.load(links, {
+        baseURI: 'http://ncode.syosetu.com/a/b/c',
```
Comment (fb55) [inline test/api/url.resolve.js]: This should be replaced with `example.com`

**Comment by fb55** [comment]
Would be great if someone could pick this up. There are currently three blockers:

- Needs a rebase
- Unnecessary indirection, the `url.resolve` call can be inlined.
- Links point to a private website, should be replaced

**Comment by bluelovers** [comment]
i don't know why u not manual fix when merge this pr

**Pull request closed without merging**